extensions/vscode/src/nodeClientMain.ts

```diff
@@ -181,3 +182,27 @@ function updateProviders(client: lsp.LanguageClient) {
 		return initializeFeatures.call(client, ...args);
 	};
 }
+
+try {
+	const tsExtension = vscode.extensions.getExtension('vscode.typescript-language-features')!;
+	const readFileSync = fs.readFileSync;
+	const extensionJsPath = require.resolve('./dist/extension.js', { paths: [tsExtension.extensionPath] });
+
+	// @ts-expect-error
+	fs.readFileSync = (...args) => {
+		if (args[0] === extensionJsPath) {
+			// @ts-expect-error
+			let text = readFileSync(...args) as string;
+
+			// patch jsTsLanguageModes
+			text = text.replace('t.$u=[t.$r,t.$s,t.$p,t.$q]', s => s + '.concat("vue")');
+
+			// patch isSupportedLanguageMode
+			text = text.replace('s.languages.match([t.$p,t.$q,t.$r,t.$s]', s => s + '.concat("vue")');
+
+			return text;
+		}
+		// @ts-expect-error
+		return readFileSync(...args);
+	};
```

Comment:
Have you tried submitting some fix for that in VSCode repo? This looks like something that nobody should be allowed to do ;)

I suppose it would likely make sense if it would also consider `languages` declared by a plugin when showing the status field?

Reply:
I'll leave it to VSCode team to decide how to implement it, please track https://github.com/microsoft/vscode/issues/200511